### PR TITLE
Add support for building model with AzureBlobContentSource

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,9 @@
 - Renamed `getValue*` accessor methods on `DocumentField` to `getValueAs*`
 - Renamed `DocTypeInfo` model to `DocumentTypeDetails`
 - Renamed `docTypes` property on `DocumentModelDetails` model to `documentTypes`
+- Renamed `trainingFilesUrl` to `blobContainerUrl` in `beginBuildModel` method
+- Changed method `beginBuildModel(String, DocumentModelBuildMode, BuildModelOptions)` to `beginBuildModel(String, DocumentModelBuildMode, String, BuildModelOptions)`
+- Removed `prefix` property on model `BuildModelOptions`
```
Comment (kinelski) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:24]: From my understanding, Java also has a simpler overload that takes only the document and the build mode, is that right?
```java
beginBuildModel(String, DocumentModelBuildMode)
```

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -4,9 +4,9 @@
 
 ### Features Added
 - Add a constructor to create an instance of `CopyAuthorization` model.
+- Added BinaryData support to document analysis client using `beginAnalyzeDocument(String modelId, BinaryData document, long length)`
 
 ### Breaking Changes
```
Comment (kinelski) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:9]: Not related to this PR, but in the changelog you have this entry:
> Renamed `DocumentModelDetails` model to `DocumentModelInfo`

Should it be the other way around?

**Review by kinelski — APPROVED** [review]
LGTM! Will soon do the same in .NET.

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md (hunk 1 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:24]: Yes, it does. The max overload after this should have the options and prefix

**Pull request merged**